plugins/foreman_remote_execution/0.0/index.md

```diff
@@ -0,0 +1,373 @@
+execution.
+
+![Job detail 1](/plugins/foreman_remote_execution/{{page.version}}/job_detail_1.png)
+
+In overview tab you can see what query was used to find the hosts and
+how many hosts were found using it. Below is the evaluated template in
+generic form. If there were some fact or parameter inputs used, you
+will see only placeholders for them since they are specific to each
+host. User inputs are evaluated since they are same for all hosts. A
+chart is showing the status of the job on all targets.
+
+From this screen you can even cancel the job if it's not finished yet.
+First cancel attempt will try to cancel the run on Smart proxy. If
+proxy does not respond, you can hit the cancel button second time
+which forces the Job to be cancelled on Foreman (while it still might
+be running on Smart proxy). Last Job Task takes you to Foreman Tasks
+that represents this Job.
+
+In Hosts tab you can see all hosts targeted by this execution, status
+of the job on them and link to Foreman Task that represent the
+execution for a particular host. Note that this task is a subtask of a
+whole Job task.
+
+
+## 3.4 Jobs list
+
+You can find all Jobs when you navigate to Monitor -> Jobs. A table
+lists all jobs from history to future. You can search jobs by **Job
+name** with a queries like
+
+    job_name ~ set*
+
+This example would find all executions that run job which names starts
+with "set".
+
+![Job invocatios list](/plugins/foreman_remote_execution/{{page.version}}/job_invocations.png)
+
+# 4. Advanced topics
+
+In this section we try to explain how do things work behind the scene.
+
+## 4.1 Determining the smart proxy for host
+
+In order to perform the job on a host, there needs to be a Smart proxy
+available to do the job. Also, not all proxies can be used for all
+hosts in the infrastructure (due to possible network isolation).
+
+For now, we search for the proxy among the proxies that are assigned
+to the host: either directly (e.g. puppet master…) or via subnets
+(e.g. dhcp…). The reason for this is to make the initial setup for the
+remote execution plugin as easy as possible.
+
+In case of a fresh infrastructure, where no proxies are assigned to
+the hosts whatsoever, one can enable **remote_execution_global_proxy**
+setting. When turned on, the algorithm will fallback to the first
+available Smart proxy with required remote execution feature. This is
+useful for simpler infrastructure, as well as for testing purposes.
+
+In future, it's possible there will be more sophisticated mechanism
+available for determining the Smart proxy to use, especially in the
+multi-providers scenario.
+
+## 4.1 SSH authentication and keys
+
+The Smart proxy needs to be provided with a key to be used against the
+remote hosts (see
+[Installation instructions](plugins/foreman_remote_execution/{{page.version}}/index.html#2.Installation)
+for details). The remote hosts need to be configured to accept the
+private key that the smart proxy is using.
```

Comment:
we expect to get more information here about the authentication as we add it so it will contain more detail than installation (which should remain simple)